lib/realtime/broadcast_changes/handler.ex

```diff
@@ -0,0 +1,205 @@
+        {:stop, error}
+    end
+  end
+
+  @spec name(Tenant.t()) :: term()
+  def name(%Tenant{external_id: tenant_id}) do
+    {:via, Registry, {@registry, tenant_id}}
+  end
+
+  @spec supervisor_spec(Tenant.t()) :: term()
+  def supervisor_spec(%Tenant{external_id: tenant_id}) do
+    {:via, PartitionSupervisor, {Realtime.BroadcastChanges.Handler.DynamicSupervisor, tenant_id}}
+  end
+
+  @impl true
+  def call(message, metadata) when is_write(message) do
+    %{tenant_id: tenant_id} = metadata
+    %Write{message: message} = parse(message)
+
+    case Registry.lookup(@registry, tenant_id) do
+      [{pid, _}] ->
+        message |> decode_message() |> then(&send(pid, &1))
+        :noreply
+
+      _ ->
+        Logger.error("Unable to find BroadcastChanges for tenant: #{tenant_id}")
+        :shutdown
+    end
+  end
+
+  def call(message, _metadata) when is_keep_alive(message) do
+    %KeepAlive{reply: reply, wal_end: wal_end} = parse(message)
+    wal_end = wal_end + 1
+
+    message =
+      case reply do
+        :now -> standby_status(wal_end, wal_end, wal_end, reply)
+        :later -> hold()
+      end
+
+    {:reply, message}
+  end
+
+  def call(msg, state) do
+    Logger.warning("Unknown message received: #{inspect(%{msg: parse(msg), state: state})}")
+    :noreply
+  end
+
+  @impl true
+  def handle_info(%Decoder.Messages.Relation{} = msg, state) do
+    %Decoder.Messages.Relation{id: id, namespace: namespace, name: name, columns: columns} = msg
+    %{relations: relations} = state
+    relation = %{name: name, columns: columns, namespace: namespace}
+    relations = Map.put(relations, id, relation)
+    {:noreply, %{state | relations: relations}}
+  end
+
+  def handle_info(%Decoder.Messages.Insert{} = msg, state) do
+    %Decoder.Messages.Insert{relation_id: relation_id, tuple_data: tuple_data} = msg
+    %{buffer: buffer, relations: relations} = state
+
+    case Map.get(relations, relation_id) do
+      %{columns: columns} ->
+        to_broadcast =
+          tuple_data
+          |> Tuple.to_list()
+          |> Enum.zip(columns)
+          |> Map.new(fn
+            {nil, %{name: name}} -> {name, nil}
+            {value, %{name: name, type: "jsonb"}} -> {name, Jason.decode!(value)}
+            {value, %{name: name, type: "bool"}} -> {name, value == "t"}
+            {value, %{name: name}} -> {name, value}
+          end)
+
+        payload = Map.get(to_broadcast, "payload")
+
+        case payload do
+          nil ->
+            {:noreply, state}
+
+          payload ->
+            id = Map.fetch!(to_broadcast, "id")
+
+            to_broadcast =
+              %{
+                topic: Map.fetch!(to_broadcast, "topic"),
+                event: Map.fetch!(to_broadcast, "event"),
+                private: Map.fetch!(to_broadcast, "private"),
+                payload: Map.put(payload, "id", id)
+              }
+
+            buffer = Enum.reverse([to_broadcast | buffer])
```

Comment:
You should accumulate them all and only reverse at the end, when you emit it during the commit.